lifetimes/generate_data.py

```diff
@@ -112,7 +179,7 @@ def pareto_nbd_model(T, r, alpha, s, beta, size=1):
             next_purchase_in = stats.expon.rvs(scale=1. / l)
 
         times = np.array(times).cumsum()
-        df.iloc[i] = len(times), np.max(times if times.shape[0] > 0 else 0), T[i], l, mu, time_of_death > T[i], i
+        df.iloc[i] = len(set(int(t) for t in times)), np.max(times if times.shape[0] > 0 else 0), T[i], l, mu, time_of_death > T[i], i
```

Comment:
This is using pure Python, which is going to be slower. What about something like:
```python
np.unique(np.array(times).astype(int)).shape[0]
```
that's about 4 times faster